pandas/core/internals/concat.py

```diff
@@ -443,7 +443,10 @@ def _is_uniform_join_units(join_units: List[JoinUnit]) -> bool:
     #  cannot necessarily join
     return (
         # all blocks need to have the same type
-        all(type(ju.block) is type(join_units[0].block) for ju in join_units)
+        all(
+            type(ju.block) is type(join_units[0].block)  # noqa: E721
```

Comment:
instead use `instance`

Reply:
Thanks, I wasn't sure in this in this case if the *exact* type was actually important.  Will update.